pkg/controller/common/manage_children.go

```diff
@@ -286,7 +306,12 @@ func updateChildren(client *dynamicclientset.ResourceClient, updateStrategy Chil
 			obj.SetOwnerReferences(ownerRefs)
 
 			if _, err := client.Namespace(ns).Create(context.TODO(), obj, metav1.CreateOptions{}); err != nil {
-				errs = append(errs, err)
+				if apierrors.IsAlreadyExists(err) {
+					// Swallow the error since there's no point retrying if the child already exists
+					logging.Logger.Info("Failed to create child, child object already exists", "parent", parent, "child", obj)
```

Comment:
unsure here, it might be, if the already created differs from the one we would like to create

Reply:
To enter this else block, it means that we did not observe that the resource existed in the cluster.
https://github.com/metacontroller/metacontroller/blob/master/pkg/controller/common/manage_children.go#L202

Since the resource does not exist, if the create fails it means that the resource exists in the cluster but then we did not find in the cache yet. So, once the sharedInformer watch gets the latest from the cluster it should trigger this and not enter the else block again and will be handled by the update logic since `oldObj` is not nil.

Reply:
Yes, your'e right